### PR TITLE
Add the ability to supply a custom render function

### PR DESCRIPTION
**Pull request by rsnyman**
An application I'm working needs to render custom components, so this PR implements an optional custom render function.

BTW, I added a test, but I could never get the tests to run, there were numerous errors outside of the actual tests.

### CODE REVIEW

File: src/OAuth2Login.jsx [hunk 1]
```diff
@@ -82,15 +82,19 @@ class OAuth2Login extends Component {
...
 
   render() {
-    const { className, buttonText, children } = this.props;
-    const attrs = { onClick: this.onBtnClick };
+    const { className, buttonText, children, render } = this.props;
 
-    if (className) {
-      attrs.className = className;
+    if (render) {
+        return render({className, buttonText, children, onClick: this.onBtnClick});
```
Comment (bhubr) [inline src/OAuth2Login.jsx:88]: I'd put a space after the opening brace & before the closing one (sorry for being picky 😅 )

File: src/OAuth2Login.jsx [hunk 2]
```diff
@@ -82,15 +82,19 @@ class OAuth2Login extends Component {
...
+    }
+    else {
```
Comment (bhubr) [inline src/OAuth2Login.jsx:90]: the preceding `return` makes the else block useless imho

**Review by bhubr — CHANGES_REQUESTED** [review]
Thank you for your PR!
I'd just remove the else block. Otherwise I'm fine with these changes, it is indeed a nice addition!

File: src/OAuth2Login.jsx (hunk 1 above)
Comment (rsnyman) [reply src/OAuth2Login.jsx]: Not a problem! We all have our styles. Fixed.

**Comment by rsnyman** [comment]
@bhubr  Sorry I didn't get back to this sooner, I didn't see your replies till today! I've fixed the issues you requested and pushed up a new version.

**Review by bhubr — APPROVED** [review]

**Comment by bhubr** [comment]
Thanks for your work, merging it right now!

**Comment by bhubr** [comment]
@rsnyman I forgot to suggest updating the version number, so I'll do it myself, and mention you in the contributors in the readme (and republish the package on npm)

**Comment by rsnyman** [comment]
Thank you!

**Pull request merged**